### PR TITLE
 Add test case for `AbstractWasmMetaDataHandler`

### PR DESCRIPTION
**Pull request by Divyansh200102**
#5427

Make sure that:

- [x] You have read the [contribution guidelines](https://shenyu.apache.org/community/contributor-guide).
- [x] You submit test cases (unit or integration tests) that back your changes.
- [x] Your local test passed `./mvnw clean install -Dmaven.javadoc.skip=true`.

### CODE REVIEW

File: shenyu-plugin/shenyu-plugin-wasm-base/src/test/java/org/apache/shenyu/plugin/wasm/base/handler/AbstractWasmMetaDataHandlerTest.java [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+         * The rpc type test.
+         */
+
+    @Test
+    public void rpcTypeTest() {
+        assertEquals("SHENYU_TEST", metaDataHandler.rpcType());
+        assertEquals("SHENYU_TEST", testWasmMetaDataHandler.rpcType());
+    }
+
+    static class TestWasmMetaDataHandler implements MetaDataHandler {
```
Comment (loongs-zhang) [inline shenyu-plugin/shenyu-plugin-wasm-base/src/test/java/org/apache/shenyu/plugin/wasm/base/handler/AbstractWasmMetaDataHandlerTest.java:98]: Need to inherit `AbstractWasmMetaDataHandler`, now this implementation will not load&call the WASM function.

**Comment by Divyansh200102** [comment]
While updating the `lib.rs` of this branch this error seems to be coming it seems to be a function signature mismatch: of the `remove` function could you please tell how am I supposed to be resolving this
```powershell
  = note: rust-lld: error: function signature mismatch: remove
          >>> defined as (i32) -> i32 in C:\Users\Divyansh\.rustup\toolchains\stable-x86_64-pc-windows-msvc\lib\rustlib\wasm32-wasi\lib\self-contained\libc.a(posix.o)
          >>> defined as (i64) -> void in C:\Users\Divyansh\Documents\GitHub\shenyu\shenyu-plugin\shenyu-plugin-wasm-base\src\test\rust-shenyu-wasm-meta-data-handler-plugin\target\wasm32-wasi\release\deps\rust_shenyu_wasm_meta_data_handler_plugin.rust_shenyu_wasm_meta_data_handler_plugin.3bb942617fe0a750-cgu.0.rcgu.o


error: could not compile `rust-shenyu-wasm-meta-data-handler-plugin` (lib) due to previous error                                                                                   


```

**Review by loongs-zhang — CHANGES_REQUESTED** [review]
Please fix CI.

**Comment by Divyansh200102** [comment]
> Please fix CI.

Can you please take a look at https://github.com/apache/shenyu/pull/5452#issuecomment-1967764377

**Comment by Divyansh200102** [comment]
@loongs-zhang An out of the context question but if I want to get my GSOC proposal reviewed by you can I contact you on this mail id [1936978077@qq.com](mailto:1936978077@qq.com) ?

**Comment by Divyansh200102** [comment]
@loongs-zhang I know you have a busy schedule might have a busy schedule right now but this is my first draft of GSOC proposal for  Apache Shenyu kitex plugin.Could you please take a look at this and give some feedback?[Proposal Link](https://docs.google.com/document/d/1XixB0c96-NcCPgE6h-5zmpTZjHvyFPMjQMkm_AfsbkA/edit?usp=sharing)

**Comment by loongs-zhang** [comment]
There is indeed a conflict in the method name, I changed them to `handleMetaData` and `removeMetaData`.

**Comment by Divyansh200102** [comment]
> There is indeed a conflict in the method name, I changed them to `handleMetaData` and `removeMetaData`.

I'll work on it could you please take a look at https://github.com/apache/shenyu/pull/5452#issuecomment-1974790313

**Comment by Divyansh200102** [comment]
> There is indeed a conflict in the method name, I changed them to `handleMetaData` and `removeMetaData`.

If time allows it could you please take a look at https://github.com/apache/shenyu/pull/5452#issuecomment-1974790313

**Comment by loongs-zhang** [comment]
At present, the official GSOC has not yet announced the selected ideas.

**Comment by loongs-zhang** [comment]
see https://github.com/apache/shenyu/pull/5464

**Comment by Divyansh200102** [comment]
> At present, the official GSOC has not yet announced the selected ideas.

Please correct me if I am wrong but here it seems that the GSOC  has released the ideas list of Apache foundation and kitexPlugin is written as one of the ideas.
image

image

**Comment by loongs-zhang** [comment]
That's just an idea collection form, informal.

**Comment by Divyansh200102** [comment]
> That's just an idea collection form, informal. Based on last year's experience, the official announcement will not be published until mid April.

But according to the official GSOC 2024 timeline the contributor's proposal period is from March 18 to April 2 so if I am not wrong  contributors like us  have to submit the proposal's based on the ideas list till deadline which is April 2 and after the submitting the proposals the accepted project will be announced on May 2 ?But we have to prepare the proposal first am I wrong? Can you please clarify?

GSOC 2024 timeline link: [https://developers.google.com/open-source/gsoc/timeline#march_18_-_1800_utc](https://developers.google.com/open-source/gsoc/timeline#march_18_-_1800_utc)
```
February 6 - 20
Google program administrators review organization applications
February 21 - 18:00 UTC
List of accepted mentoring organizations published
February 22 - March 18
Potential GSoC contributors discuss application ideas with mentoring organizations
March 18 - 18:00 UTC
GSoC contributor application period begins
April 2 - 18:00 UTC
GSoC contributor application deadline
April 24 - 18:00 UTC
GSoC contributor proposal rankings due from Org Admins
May 1 - 18:00 UTC
Accepted GSoC contributor projects announced
May 1 - 26
Community Bonding Period | GSoC contributors get to know mentors, read documentation, get up to speed to begin working on their projects
May 27
Coding officially begins!
```

**Comment by Divyansh200102** [comment]
@loongs-zhang Can you please clarify https://github.com/apache/shenyu/pull/5452#issuecomment-1975050756

**Comment by Divyansh200102** [comment]
@loongs-zhang I know you are busy but if you could clarify https://github.com/apache/shenyu/pull/5452#issuecomment-1975050756 it would  be a great help.

**Comment by Divyansh200102** [comment]
> That's just an idea collection form, informal. Based on last year's experience, the official announcement will not be published until mid April.

Since the application period is currently ongoing (18 march-2 April) I as a  contributor have to submit a proposal [[GSOC Timeline]](https://developers.google.com/open-source/gsoc/timeline#march_18_-_1800_utc) .I have prepared a proposal for Apache Shenyu KitexPlugin .Could you please take a look?
 
Proposal Link: https://docs.google.com/document/d/1XixB0c96-NcCPgE6h-5zmpTZjHvyFPMjQMkm_AfsbkA/edit?usp=sharing

**Pull request closed without merging**